src/AspNet.Security.OAuth.Nextcloud/NextcloudAuthenticationOptions.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Linq;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json.Linq;
+using static AspNet.Security.OAuth.Nextcloud.NextcloudAuthenticationConstants;
+
+namespace AspNet.Security.OAuth.Nextcloud
+{
+    public class NextcloudAuthenticationOptions : OAuthOptions
+    {
+        public NextcloudAuthenticationOptions()
+        {
+            ClaimsIssuer = NextcloudAuthenticationDefaults.Issuer;
+            CallbackPath = new PathString(NextcloudAuthenticationDefaults.CallbackPath);
+
+            ClaimActions.MapCustomJson(ClaimTypes.NameIdentifier, user =>
+            {
+                return user["ocs"]?["data"]?.Value<string>("id");
+            });
+
+            ClaimActions.MapCustomJson(ClaimTypes.Name, user =>
+            {
+                return user["ocs"]?["data"]?.Value<string>("id");
```

Comment:
An `id` isn't a _"name"_, so this doesn't look like the right fit for this claim and probably should be removed.

Reply:
Nextcloud returns the username as ID. 
Check test/AspNet.Security.OAuth.Providers.Tests/Nextcloud/bundle.json#32

Reply:
But a user name isn't a name. My GitHub user name is `martincostello`, but that isn't my _name_, my name is _Martin Costello_.

Reply:
i know but i looked around and saw the same structure in other providers. So sticked to that structure.
Example: 
https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/2be2d28c45e3b619c9e5017acc2ed4afb084a62a/src/AspNet.Security.OAuth.Discord/DiscordAuthenticationOptions.cs#L30
https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/src/AspNet.Security.OAuth.GitHub/GitHubAuthenticationOptions.cs#L31
https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/src/AspNet.Security.OAuth.Twitch/TwitchAuthenticationOptions.cs#L38

Reply:
I think it's better to favour correctness - users can easily add the custom claims mapping themselves if they want to make the user name the name.